### PR TITLE
Ignore files from .gitignore during format

### LINKED ISSUE #542: Ignore gitignored files when formatting a directory
If we run `gleam format` in a project it should not format the contents of the `_build` directory (or any other gitignored directories).

If there is no gitignore we should continue without filtering.

This library looks useful: https://github.com/nathankleyn/gitignore.rs

### PR DESCRIPTION
**Pull request by shritesh**
Closes #542

This is the most readable I could make the code given `gitignore`'s API. Suggestions welcome

### CODE REVIEW

File: src/format/command.rs [hunk 1]
```diff
@@ -96,8 +96,23 @@ pub fn read_and_format_paths(files: Vec<String>) -> Result<Vec<Formatted>, Error
         })?;
 
         if path.is_dir() {
-            for path in file::gleam_files(&path).into_iter() {
-                formatted_files.push(format_file(path)?);
+            let mut gitignore_file = path.clone();
+            gitignore_file.push(".gitignore");
```
Comment (lpil) [inline src/format/command.rs:100]: Can we use `join` rather than `clone` + `push`?

File: src/format/command.rs [hunk 2]
```diff
@@ -96,8 +96,23 @@ pub fn read_and_format_paths(files: Vec<String>) -> Result<Vec<Formatted>, Error
...
+                        !excluded
+                    })
+                    .into_iter()
+                {
+                    formatted_files.push(format_file(path)?);
+                }
+            } else {
+                for path in file::gleam_files(&path).into_iter() {
+                    formatted_files.push(format_file(path)?);
+                }
```
Comment (lpil) [inline src/format/command.rs:115]: Looks good! Perhaps we could reshuffle it a bit like this? (pseudocode)

```rust
let ignore = gitignore::File::new(&gitignore_file)
let not_ignored = |f| match ignore {
    Ok(ignore) -> !ignore.is_excluded(f).unwrap_or(false),
    Err(_) -> true,
};

for path in file::gleam_files(&path)
    .filter(not_ignored)
    .into_iter()
{
    formatted_files.push(format_file(path)?);
}
```

File: src/format/command.rs (hunk 1 above)
Comment (shritesh) [reply src/format/command.rs:100]: Definitely! I didn't know .join was a thing!!

**Comment by shritesh** [comment]
Closing because of https://github.com/nathankleyn/gitignore.rs/issues/15

**Comment by lpil** [comment]
Thank you, sorry for putting you down a bad path there!

**Comment by shritesh** [comment]
No problem!!

**Pull request closed without merging**